doc/test-wrapper.md

```diff
@@ -0,0 +1,123 @@
+Test Wrapper for Interactive Environments
```

Comment:
nit: Wouldn't it make more sense to put this file in the `test/` directory? I feel like we already have some documentation there and it could be referenced from the existing docs.